### PR TITLE
feat(pose_estimator_arbiter): add pose_estimator_arbiter with simple switching rule

### PR DESCRIPTION
**Pull request by KYabuuchi**
## Description

This PR implement a mechanism to launch multiple pose_estimators, and allow for pausing and resuming specific ones according to the situation. 
**This PR introduces only one rule that will always enable all pose_estimators.**

[The package README](https://github.com/KYabuuchi/autoware.universe/tree/feat/pose_estimator_arbiter_with_only_simple_rule/localization/pose_estimator_arbiter) describes the purpose in detail.

### Related links
This PR is derived from https://github.com/autowarefoundation/autoware.universe/pull/5846 & https://github.com/autowarefoundation/autoware.universe/pull/6089

Diff from the original PR(https://github.com/autowarefoundation/autoware.universe/pull/5846):
- Changed the default switching rule
- Created `example_rule` directory

Diff from the second PR(https://github.com/autowarefoundation/autoware.universe/pull/6089):
- Moved `vector_map_based_rule` into example_rule/
- **Changed the default rule to always enable all estimators**
- Simplified example_rule to clarify roles

## Related links

[[Proposal] Runtime Localization Module Switching for Multi-Pose Estimators #3878](https://github.com/orgs/autowarefoundation/discussions/3878)

## Tests performed

I have confirmed that ndt/yabloc/eagleye/artag is available as before with `logging_simulator.launch.xml` .
Also, I confirmed that by specifying multiple pose_estimators at startup, pose_estimator_arbiter is launched and arbitrates the pose_estimators according to the situation.

### sample data

* [sample map & rosbag](https://drive.google.com/file/d/1MxLo1Sw6PdvfkyOYf_9A5dZ9uli1vPvS/view)

The rosbag is simulated data created by [AWSIM](https://tier4.github.io/AWSIM/).
The map is an edited version of the original map data published on the AWSIM documentation page to make it suitable for multiple pose_estimators.

### how to launch 

When start a single pose_estimator:
```shell
ros2 launch autoware_launch logging_simulator.launch.xml \
    map_path:=$HOME/MAP vehicle_model:=sample_vehicle sensor_model:=awsim_sensor_kit \
    pose_source:=ndt # or yabloc, eagleye or artag
```

When start multiple pose_estimator:
```bash
ros2 launch autoware_launch logging_simulator.launch.xml \
    map_path:=$HOME/MAP vehicle_model:=sample_vehicle sensor_model:=awsim_sensor_kit \
    pose_source:=ndt_yabloc_eagleye_artag # or ndt_yabloc, eagleye_artag and so on
```

### expected behavior
When start a single pose_estimator, it launches the same nodes as before.

When start multiple pose_estimator, it works as follows.

https://github.com/autowarefoundation/autoware.universe/assets/24854875/d4d48be4-748e-4ffc-bde2-d498911ed3a1

* The sample map has areas that do not contain point clouds as shown in the following figure. When running with multiple pose_estimators, no fatal breakdowns will occur.

drawing

> [!TIP]
> We can visualize the debug prints of pose_estimator_arbiter at any time with the following command
> (I recommend enabling logging, because with the default rviz settings, we do not know which pose_estimator is running.)
```
ros2 service call /localization/pose_estimator_arbiter/config_logger \
    logging_demo/srv/ConfigLogger \
   '{logger_name: localization.pose_estimator_arbiter , level: debug}'
```

> [!CAUTION]
> Eagleye requires a GNSS doppler, but AWSIM does not generate it. So I have stored my pseudo-generated dopplers in rosbag. Because of that, Eagleye's estimated position is not accurate.

## Notes for reviewers

> [!TIP]
> If you want to create an rviz screen like the demonstration video, please visualize the following topics.
> `/localization/pose_estiamtor/points_aligned`
> `/localization/pose_estimator/yabloc/image_processing/lanelet2_overlay_image`
> `/localization/pose_estimator/yabloc/pf/predicted_particles_marker`
> `/localization/pose_estimator/ar_tag_based_localizer/debug/image`
> ~`/localization/eagleye/pose_with_covariance/to_relay`~
> `/localization/pose_estimator/eagleye/pose_with_covariance/to_relay`

Sorry for the large number of commits.

## Interface changes

The default behavior remains the same. 
Users can now specify multiple pose_estimators for `pose_source` . By default, ndt is launched.

## Effects on system behavior

The default behavior remains the same. 
Users can now specify multiple pose_estimators for `pose_source` . By default, ndt is launched.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by SakodaShintaro** [comment]
@KYabuuchi 
Sorry, there seems to be a conflict.

**Comment by KYabuuchi** [comment]
@SakodaShintaro Sorry, I fixed it.

**Comment by KYabuuchi** [comment]
# For reviewers
Recently, the PR https://github.com/autowarefoundation/autoware.universe/pull/6091 is merged.  When checking this PR, please note that the following PRs are reflected :pray:
* sample_sensor_kit https://github.com/autowarefoundation/sample_sensor_kit_launch/pull/85
* autoware_launch https://github.com/autowarefoundation/autoware_launch/pull/796
* awsim_sensor_kit https://github.com/RobotecAI/awsim_sensor_kit_launch/pull/14

**Review by SakodaShintaro — APPROVED** [review]
Looks good to me

Although I have not conducted proper measurements, I think this pull request may also contribute to faster launch of Autoware.
(This is because multi-loading of eagleye and pose_initializer has been improved)

File: localization/pose_estimator_arbiter/src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+      continue;
+    }
+
+    if (toggle_list.at(s.first)) {
+      s.second->enable();
+    } else {
+      s.second->disable();
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (KYabuuchi) [reply localization/pose_estimator_arbiter/src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp:151]: Thanks for your advice! I employed your suggestion https://github.com/autowarefoundation/autoware.universe/pull/6144/commits/c97b13b33b0e9fc7eaeea03317bec4ac1ca22015

File: localization/pose_estimator_arbiter/src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    // Publish visualization_msgs::MarkerArray for debug
+    pub_debug_marker_array_->publish(switch_rule_->debug_marker_array());
+
+  } else {
+    RCLCPP_WARN_STREAM(
+      get_logger(), "switch_rule is not activated. Therefore, enable all pose_estimators");
+    toggle_all(true);
+  }
+
+  //
```
[the remark this answers is not in the document]
Comment (KYabuuchi) [reply localization/pose_estimator_arbiter/src/pose_estimator_arbiter/pose_estimator_arbiter_core.cpp]: Appreciate you noticing that. I had forgotten to add comments. 
 Fixed in https://github.com/autowarefoundation/autoware.universe/pull/6144/commits/3f5e108d47cfbe091378edc9dd5ec6602f544050

**Pull request merged**